iModelCore/ECDb/PublicAPI/ECDb/ECSqlStatement.h

```diff
@@ -249,6 +249,11 @@ struct EXPORT_VTABLE_ATTRIBUTE ECSqlStatement
         //! @return Parameter binder
         ECDB_EXPORT IECSqlBinder& GetBinder(int parameterIndex);
 
+        //! Gets BinderInfo for the binder associated with the parameter at the specified index.
+        //! @param[in] parameterIndex Parameter index.
+        //! @return Parameter BinderInfo. If there is no valid binder on that index the it will still return BinderInfo with Binder type NoopECSqlBinderType
+        ECDB_EXPORT BinderInfo const& GetBinderInfo(int parameterIndex) { return GetBinder(parameterIndex).GetBinderInfo(); }

```

Comment:
From the statement level we get a const reference so that we can just read values and modify values but from the binder level I have exposed normal reference so that we can also modify values if required and I have also done so in ECSqlBinder.cpp

Reply:
oh, I see. But that seems like it could be problematic, too. You could just make the fields on binder mutable, that BinderInfo object is bypassing the const mechanism anyways

Reply:
yes I have made it const everywhere